frontend/src/plaquettes/Footprint.js

```diff
@@ -3,123 +3,194 @@ import Plaquette from './Plaquette';
+    this.unselectQubitsButton.visible = false;
+    this.selectQubitsButton.visible = true;
+    this.createPlaquetteButton.visible = false;
+    this.workspace.addChild(this.selectQubitsButton);
+    this.workspace.addChild(this.unselectQubitsButton);
+    this.workspace.addChild(this.createPlaquetteButton);
+  };
+
+  /**
+   * Select a qubit from the footprint
+   * @function selectQubit
+   * @param {Event} e - The click event
+   * @returns {void}
+   */
+  selectQubit = (e) => {
+    // Check if the click was on a qubit
+    const canvasRect = this.app.view.getBoundingClientRect(); // Get canvas position
+
+    // Calculate the relative click position within the canvas
+    const relativeX = e.clientX - canvasRect.left;
+    const relativeY = e.clientY - canvasRect.top;
+    // Get all the qubits
+    const qubits = this.workspace.children.filter(
+      (child) => child.isQubit === true
+    );
+    const qubit = qubits.find(
+      // Find the qubit that was clicked
+      (q) => q.checkHitArea(relativeX, relativeY) === true
+    );
+    if (!qubit && !(qubit?.isQubit === true)) return; // Check that the qubit exists
+    // Check that the qubit is not already selected
+    if (this.selectedQubits.includes(qubit)) {
+      // Remove the qubit from the selected qubits
+      this.selectedQubits = this.selectedQubits.filter((q) => q !== qubit);
+      return;
+    }
+    qubit.changeColor('red');
+    this.selectedQubits.push(qubit);
+    // Change the color of the qubit
+    // Check if the selected qubits are 3 or more
+    if (this.selectedQubits.length > 2) {
+      // Show the button
+      this.selectQubitsButton.visible = false;
+      this.createPlaquetteButton.visible = true;
+    }
+  };
+
+  /**
+   * Create the plaquette from the selected qubits and assign it to the footprint
+   * @function createPlaquette
+   * @returns {void}
+  */
+  createPlaquette = () => {
+    // Check that the selected qubits are part of the footprint area
+    if (this.selectedQubits.length < 3) {
+      notification(this.app, 'Plaquette requires 3+ qubits');
+      return;
+    }
+    // Render the plaquette
+    const plaquette = new Plaquette(this.selectedQubits, this.workspace, this.app);
+    // Add the plaquette to the tile container
+    this.container.addChild(plaquette);
+    // Remove seleected qubits from the footprint qubits, so they can be used again
+    // For each qubit, remove the text
+    this.selectedQubits.forEach((qubit) => {
+      qubit.removeChildren();
+    });
+    // Clear the selected qubits
+    this.selectedQubits = [];
+    // Notify the user that the plaquette has been created
+    notification(this.app, 'Plaquette created');
+    this.selectQubitsButton.visible = true;
+    this.createPlaquetteButton.visible = false;
+    this.unselectQubitsButton.visible = false;
+  };
+
+  /**
+   * Render the footprint
+   * @function renderFootprintControlButtons
+   * @returns {void}
+   */
+  renderFootprintControlButtons() {
```

Comment:
A better name would be something like `updateVisiblityAndEventListeners()` since rendering is not the job of a `Container` per se